### PR TITLE
Nodejs week2

### PR DESCRIPTION
**Pull request by myselfankitaa**
Write command:

1. get/search
2. get/document/:id
3. post/search

### CODE REVIEW

File: nodejs/week2/app.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+      String(element.value).toLowerCase().includes(query.toLowerCase()) ||
+      String(element.description).toLowerCase().includes(query.toLowerCase())
+  );
+
+  if (matchingData.length > 0) {
+    res.send(matchingData);
+    found = true;
+  }
+  if (!found) {
+    res.status(404).send("There is an error in fetching data");
```
Comment (babak-f) [inline nodejs/week2/app.js]: The error code 404 is for when a document, object etc is not found. Here you could use for example 500.
See https://developer.mozilla.org/en-US/docs/Web/HTTP/Status

File: nodejs/week2/app.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+  }
+});
+
+app.get("/document/:id", (req, res) => {
+  const param = req.params.id;
+  try {
```
Comment (babak-f) [inline nodejs/week2/app.js:52]: I can see that this function has error handling with try-catch, which is good, but some other functions don't have it. It's better to be consistent and implement it for all of them 😉

**Review by babak-f — APPROVED** [review]
All good and good error handling. I left a couple of minor comments.

File: nodejs/week2/app.js (hunk 1 above)
Comment (myselfankitaa) [reply nodejs/week2/app.js]: Ok, Thanks noted I will implement accordingly

Comment (myselfankitaa) [reply nodejs/week2/app.js]: Ok thanks for sharing link for more clarity. I have updated the file accordingly

**Pull request merged**